app/controllers/merchant_items_controller.rb

```diff
@@ -1,7 +1,8 @@
 class MerchantItemsController < ApplicationController
   def index
     @merchant = Merchant.find(params[:id])
-    @items = @merchant.items
+    @enabled_items_array = @merchant.items.enabled_items
```

Comment:
rather than send three @variables to the view, we should send only @merchant and create two merchant instance methods to be called in the view like @merchant.enabled_items and @merchant.disabled_items